compiler/luci-interpreter/src/kernels/AveragePool2D.cpp

```diff
@@ -61,11 +62,13 @@ void AveragePool2D::configure()
       computePadding(_params.stride_width, 1, input_width, _params.filter_width, output_width);
   if (input()->element_type() == DataType::U8)
   {
-    if (input()->scale() != output()->scale() || input()->zero_point() != output()->zero_point())
-    {
-      throw std::runtime_error(
-          "Quantization param for Input and output must be same(scale or zero-point)");
-    }
+    LUCI_INTERPRETER_CHECK(std::abs(output()->scale() - input()->scale()) <= 1.0e-6);
```

Comment:
Where did you get 1.0e-6 delta?
Looks like input and output scales should be totally equal.

Reply:
This is what TFLite does.
I'm ok with equality comparison too.
@jinevening what do you think?

Reply:
> @jinevening what do you think?

I think it would be safe to follow the TFLite code because we are using their kernel. :)

FYI, our quantizer does not have such a restriction (input scale == ouput scale) for average_pool.

Reply:
> FYI, our quantizer does not have such a restriction (input scale == ouput scale) for average_pool.

Is it intentional?

Reply:
Just for the case, TFLite only checks that the scales are (almost) equal, it does not make use of them otherwise.

Reply:
> Is it intentional?

I have no idea. Some restrictions may be added in the future for performance, but the current instruction model does not have such a restriction.

> Just for the case, TFLite only checks that the scales are (almost) equal, it does not make use of them otherwise.

I think that the purpose of checking the equality of scales is to avoid 'rescaling'. It seems that TFLite kernel does not perform rescaling under the assumption that the scales of input and output are the same (or almost equal).

Reply:
> I have no idea. Some restrictions may be added in the future for performance, but the current instruction model does not have such a restriction.

Would you want to remove this restriction from interpreter too (the kernel will have to be reimplemented by hand), or is it OK to go as it is for now?

Reply:
I think we can go as is for now. :)

Reply:
> I think we can go as is for now. :)

OK. I'm just a bit worried about using interpreter with models quantized with our quantizer tool.